### PR TITLE
feat(datahub-frontend): Adding basic file-based authentication to datahub-frontend

### PR DESCRIPTION
**Pull request by jjoyce0510**
**Summary**
This change introduces file-based authentication to datahub-frontend as the default mechanism deployed with DataHub. 

By default, datahub-frontend will deploy with a `user.props` file containing a single username:password pair - datahub:datahub. 

Note that this form of authentication is _basic_. The password is stored inside the `user.props` file in **plaintext**.  If you need a more secure way of configuring `datahub-frontend`, we'd recommend leveraging a custom JaaS module. 

Both the `user.props` file and the `jaas.conf` file can be customized by overriding the default files mounted to your container inside `/datahub-frontend/conf` directory. This is outlined in a new doc provided inside `docs/how/auth`. 

**Acknowledgements**
Thanks to @frsann for pioneering this solution!   

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-frontend/run/jaas.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// This is a sample JAAS config that uses the following login module
+// DummyLoginModule -- this module can work with a username (that is in user table) and any password
+
+WHZ-Authentication {
+  org.eclipse.jetty.jaas.spi.PropertyFileLoginModule sufficient
+  debug="true"
+  file="../conf/user.props";
```
Comment (frsann) [inline datahub-frontend/run/jaas.conf:7]: Newline

File: datahub-frontend/conf/jaas.conf [hunk 2]
```diff
@@ -2,5 +2,7 @@
 // DummyLoginModule -- this module can work with a username (that is in user table) and any password
 
 WHZ-Authentication {
-  security.DummyLoginModule sufficient;
-};
+  org.eclipse.jetty.jaas.spi.PropertyFileLoginModule sufficient
+  debug="true"
+  file="/datahub-frontend/conf/user.props";
+};
```
Comment (frsann) [inline datahub-frontend/conf/jaas.conf:8]: newline nit

File: datahub-frontend/run/jaas.conf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// This is a sample JAAS config that uses the following login module
+// DummyLoginModule -- this module can work with a username (that is in user table) and any password
```
Comment (frsann) [inline datahub-frontend/run/jaas.conf:2]:
```suggestion
// org.eclipse.jetty.jaas.spi.PropertyFileLoginModule -- this module can work with a username and any password defined in the `../conf/user.props` file
```

File: datahub-frontend/run/jaas.conf [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+
+WHZ-Authentication {
+  org.eclipse.jetty.jaas.spi.PropertyFileLoginModule sufficient
+  debug="true"
+  file="../conf/user.props";
+};
```
Comment (frsann) [inline datahub-frontend/run/jaas.conf:8]: newline

File: docs/how/auth/jaas.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+# JaaS Authentication
+
+## Overview
+
+The DataHub frontend server comes with support for plugging in [JaaS](https://docs.oracle.com/javase/7/docs/technotes/guides/security/jaas/JAASRefGuide.html) modules. 
+This allows you to use a custom authentication protocol to log your users into DataHub. 
+
+By default, we in include sample configuration of a file-based username / password authentication module ([PropertyFileLoginModule](http://archive.eclipse.org/jetty/8.0.0.M3/apidocs/org/eclipse/jetty/plus/jaas/spi/PropertyFileLoginModule.html))
+that is configured with a single username / password combination: datahub - datahub.
```
Comment (frsann) [inline docs/how/auth/jaas.md:9]: And once we have fine grained access controls, this login module allows us to define the use role as well 👍

**Review by frsann — APPROVED** [review]
Some minor nits, but this looks great!

File: docs/how/auth/jaas.md (hunk 5 above)
Comment (shirshanka) [reply docs/how/auth/jaas.md:9]: It does!

Comment (jjoyce0510) [reply docs/how/auth/jaas.md:9]: Heck yeah!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**